.editorconfig

```
@@ -0,0 +1,31 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+indent_style = space
+
+[Makefile]
+indent_style = tab
+
+[*.py]
+charset = utf-8
+indent_size = 4
+max_line_length = 88
+
+[*.js]
+charset = utf-8
+
+[*.{html,rst,md}]
+indent_size = 4
+
+[*.{js,ts,tsx,json,yml,yaml,css,scss}]
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = false
+
+# Make sure this file _doesn’t_ end with a final newline. It breaks rendering.
+[wagtail/admin/templates/wagtailadmin/shared/icon.html]
+insert_final_newline = false
```

```suggestion
[*.md]
trim_trailing_whitespace = false
```

We can probably skip that last one :) Also with this in place, I’m surprised that file doesn’t have a final newline. Double check how your editor is configured when you have the chance!